Xamarin.Forms.Platform.UAP/PopupRenderer.cs

```diff
@@ -0,0 +1,251 @@
+		XamlStyle _flyoutStyle;
+		XamlStyle _panelStyle;
+		public PopupRenderer()
+		{
+		}
+
+		void IVisualElementRenderer.SetElement(VisualElement element)
+		{
+			if (element == null)
+				throw new ArgumentNullException(nameof(element));
+
+			if (!(element is BasePopup popup))
+				throw new ArgumentNullException("Element is not of type " + typeof(BasePopup), nameof(element));
+
+			BasePopup oldElement = Element;
+			Element = popup;
+			CreateControl();
+
+			Performance.Start(out string reference);
+
+			if (oldElement != null)
+				oldElement.PropertyChanged -= OnElementPropertyChanged;
+
+			element.PropertyChanged += OnElementPropertyChanged;
+
+			OnElementChanged(new ElementChangedEventArgs<BasePopup>(oldElement, Element));
+
+			Performance.Stop(reference);
+		}
+
+		void CreateControl()
+		{
+			if (Control == null)
+			{
+				Control = new ViewToRendererConverter.WrapperControl(Element.View);
+				Content = Control;
+			}
+		}
+
+		void InitializeStyles()
+		{
+			_flyoutStyle = new XamlStyle { TargetType = typeof(FlyoutPresenter) };
+			_panelStyle = new XamlStyle { TargetType = typeof(Panel) };
+		}
+
+		protected virtual void OnElementChanged(ElementChangedEventArgs<BasePopup> e)
+		{
+			if (e.NewElement != null && !_isDisposed)
+			{
+				InitializeStyles();
+				SetEvents();
+				SetColor();
+				SetBorderColor();
+				SetSize();
+				SetLayout();
+				ApplyStyles();
+				Show();
+			}
+
+			ElementChanged?.Invoke(this, new VisualElementChangedEventArgs(e.OldElement, e.NewElement));
+		}
+
+		protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEventArgs args)
+		{
+			if (args.PropertyName == BasePopup.VerticalOptionsProperty.PropertyName ||
+				args.PropertyName == BasePopup.HorizontalOptionsProperty.PropertyName ||
+				args.PropertyName == BasePopup.SizeProperty.PropertyName ||
+				args.PropertyName == BasePopup.ColorProperty.PropertyName)
+			{
+				ConfigureControl();
+			}
+
+			ElementPropertyChanged?.Invoke(this, args);
+		}
+
+		void ConfigureControl()
+		{
+			InitializeStyles();
+			SetEvents();
+			SetColor();
+			SetBorderColor();
+			SetSize();
+			SetLayout();
+			ApplyStyles();
+		}
+
+		void SetEvents()
+		{
+			Element.Dismissed += OnDismissed;
+		}
+
+		void SetSize()
+		{
+			var standardSize = new Size { Width = _defaultSize, Height = _defaultSize / 2 };
+
+			Size currentSize = Element.Size != default(Size) ? Element.Size : standardSize;
+			Control.Width = currentSize.Width;
+			Control.Height = currentSize.Height;
+
+			_flyoutStyle.Setters.Add(new Windows.UI.Xaml.Setter(FlyoutPresenter.MinHeightProperty, currentSize.Height + (_defaultBorderThickness * 2)));
+			_flyoutStyle.Setters.Add(new Windows.UI.Xaml.Setter(FlyoutPresenter.MinWidthProperty, currentSize.Width + (_defaultBorderThickness * 2)));
+		}
+
+		void SetLayout()
+		{
+			LightDismissOverlayMode = LightDismissOverlayMode.On;
+			SetDialogPosition(Element.VerticalOptions, Element.HorizontalOptions);
```

Comment:
I am unable to get the `Flyout.Placement` to update correctly when the `VerticalOptions` or `HorizontalOptions` are updated for UWP. I've tried the following:
- Updating the `_flyoutStyle` and `_panelStyle`
- Hiding the flyout and trying to redraw it
- Nulling out the attached Flyout and re-adding it to the parent control
- Various orders of styles and placement updates

It almost appears that I need to completely dispose of the control and re-initialize it but then we risk losing state. If anyone has suggestions on how we can get this to work I am all ears.